shuffle/cli/src/shared.rs

```diff
@@ -103,6 +107,86 @@ pub fn get_shuffle_dir() -> PathBuf {
     BaseDirs::new().unwrap().home_dir().join(".shuffle")
 }
 
+pub struct DevApiClient {
+    client: Client,
+}
+
+// Client that will make GET and POST requests based off of Dev API
+impl DevApiClient {
+    pub fn new(client: Client) -> Result<Self> {
+        Ok(Self { client })
+    }
+
+    pub async fn get_account_modules(
+        &self,
+        network: &Url,
+        address: AccountAddress,
+    ) -> Result<Response> {
+        let path =
+            network.join(format!("accounts/{}/modules", address.to_hex_literal()).as_str())?;
+        Ok(self.client.get(path.as_str()).send().await?)
+    }
+
+    pub async fn get_transactions_with_hash(&self, hash: &str, network: &Url) -> Result<Response> {
```

Comment:
get_transactions_with_hash => get_transaction_by_hash

note: `transactions` indicates response is list